DOCUMENTATION.md

```diff
@@ -0,0 +1,305 @@
+
+You can of course rotate it around Z axis with `data-rotate-z` - it has exactly the same effect as `data-rotate` (these two are basically aliases).
+
+```HTML
+<div id="its-in-3d" class="step" data-x="6200" data-y="4300" data-z="-100" data-rotate-x="-40" data-rotate-y="10" data-scale="2">
+    <p>
+      <span class="have">have</span>
+      <span class="you">you</span>
+      <span class="noticed">noticed</span>
+      <span class="its">it’s</span>
+      <span class="in">in</span>
+      <b>3D<sup>*</sup></b>?
+    </p>
+    <span class="footnote">* beat that, prezi ;)</span>
+</div>
+```
+
+## CSS
+
+### 4D States (.past, .present and .future classes)
+
+The `.future` class is added to all [Step Elements](#step-element) that haven't been visited yet.
+
+**Example:**
+
+```CSS
+.future {
+  display: none;
+}
+```
+
+The `.present` class is added to the [Step Element](#step-element) that is currently at the center of the camera. This is useful to create animations inside the step once the camera navigates to it.
+
+**Example:**
+
+```CSS
+.present .rotating {
+  transform: rotate(-10deg);
+  transition-delay: 0.25s;
+}
+```
+
+The `.past` class is added to all [Step Elements](#step-element) that have been visited at least once.
+
+**Example:**
+
+```CSS
+.past {
+  display: none;
+}
+```
+
+### Current Active Step (.active class)
+
+The `.active` class is added to the [Step Element](#step-element) that is currently visible at the center of the camera.
+
+**Example:**
+
+```CSS
+.step {
+  opacity: 0.3;
+  transition: opacity 1s;
+}
+.step.active {
+  opacity: 1
+}
+```
+
+At the same time, the `impress-on-*` class is added to the body element, the class name represents the active [Step Element](#step-element) id. This allows for custom global styling, since you can't match a CSS class backwards from the active [Step Element](#step-element) to the `body`.
+
+**Example:**
+
+```CSS
+.impress-on-overview .step {
+    opacity: 1;
+    cursor: pointer;
+}
+.impress-on-step-1,
+.impress-on-step-2,
+.impress-on-step-3 {
+  background: LightBlue;
+}
+```
+
+### Progressive Enhancement (.impress-not-supported class)
+
+The `.impress-not-supported` class is added to the `body` element if the browser doesn't support the features required by impress.js to work, it is useful to apply some fallback styles in the CSS.
+
+It's not necessary to add it manually on the `body` element. If the script detects that the browser lacks important features it will add this class.
+
+It is recommended to add the class manually to the `body` element though, because that means that users without JavaScript will also get fallback styles. When impress.js script detects that the browser supports all required features, the `.impress-not-support` class will be removed from the `body` element.
+
+**Example:**
+
+```CSS
+.impress-not-supported .step {
+  display: inline-block;
+}
+```
+
+## JavaScript
```

Comment:
While usage based on default `impress()` is most common, I'd add a section explaining this "constructor" (or maybe more like a factory function) of impress.js.

`impress( rootId )` is a function that builds API for presentation with given id of root element ('impress' by default).

To avoid calling it all the time it's worth storing it in a variable:
```
var api = impress("my-root");
```

It's possible to have multiple instances of impress.js (with separate roots) on the same page:

```
var firstAPI = impress("first-root");
var secondAPI = impress("second-root");

firstAPI.init();
secondAPI.init();
```